STM32: EXTI driver cleanup

erwango opened the pull request:
`driver/interrupt_controller/exti_stm32.c` holds a rather complex code which is hard to maintain and update.

Perform some clean up:
 - Extend IRQ table concept proposed for MP1 series
 - Remove unused (because non functional) code

erwango committed:
drivers/interrupt_controller: stm32: Rework stm32_exti_enable

Function stm32_exti_enable is complex and makes it hard to extend
to new stm32 series.
When MP1 support was added, table exti_irq_table was introduced
to simplify stm32_exti_enable function.
This change extends usage of this table to other series, without
adding or removing any functionality.


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@arnop2

erwango commented:
Set as DNM waiting for additional tests

erwango commented:
Tested on all series.

mbolivar commented on drivers/interrupt_controller/exti_stm32.c:
```diff
@@ -51,7 +51,51 @@
-#if defined(CONFIG_SOC_SERIES_STM32MP1X)
+#if defined(CONFIG_SOC_SERIES_STM32F0X) || \
+    defined(CONFIG_SOC_SERIES_STM32L0X)
+IRQn_Type exti_irq_table[] = {
```
Adding const would save some RAM, no? Here and elsewhere.

erwango replied:
Sure, let me have a try. Thanks!

mbolivar reviewed (approved):
Another nice cleanup!

erwango committed:
drivers/interrupt_controller: stm32: Remove unused code

Some code in stm32 exti driver was considering exti lines number
for which no IRQ_CONNECT was available. To be clear, this code
was not functional and since no one complained it was not used.
Besides, code to take into account these "high" irq lines was
complex hard to read and review.
Simplify state of things by removing completely this part of code.
It could be put back again piece by piece when required.

This change allows to get rid of EXTI_LINES definition.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers: interupt_controller: stm32 Move exti_irq_table to const

Save few RAM bytes by declaring exti_irq_table as const.

Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
